[CodeGen] @llvm.experimental.stackmap make operands immediate

Atafid opened the pull request:
This pull request modifies the behavior of the `@llvm.experimental.stackmap` intrinsic to require that its two first operands (`id` and `numShadowBytes`) be **immediate values**. This change ensures that variables cannot be passed as two first arguments to this intrinsic.


Related Issue: https://github.com/llvm/llvm-project/issues/115733

### Testing
- Added new test cases to ensure errors are emitted for non-immediate operands.
- Ran the full LLVM test suite to verify no regressions were introduced.

Atafid committed:
Add immediate operands to llvm.experimental.stackmap

Atafid commented:
Thank you I didn't know this pattern, I used it to update my tests.
I squashed the commits into one to keep the history clean. Let me know if further changes are needed

Atafid commented:
@uweigand @phoebewang 
Hello, sorry for the cc, have you any other comments on what I wrote ? :)

uweigand commented:
The test cases look good to me now.  I don't have any additional comments.

phoebewang reviewed (approved):
Patch LG.

I didn't find a description in LangRef. Would be great if it's completed.

Atafid commented:
The documentation seems to be in `llvm/docs/StackMaps.rst`, I added a sentence to inform that the operands should be immediate.
Again I squashed the commits into one to keep the history clean.

phoebewang commented:
> The documentation seems to be in `llvm/docs/StackMaps.rst`, I added a sentence to inform that the operands should be immediate. Again I squashed the commits into one to keep the history clean.

Thanks for the update! Do you need me to merge the patch for you?

Atafid commented:
If you have no further comments, I would really appreciate it if you could merge it. Thank you! :)

phoebewang commented:
> If you have no further comments, I would really appreciate it if you could merge it. Thank you! :)

You are welcome :)
